runtime/recipe/handle.js

```diff
@@ -114,8 +114,17 @@ export class Handle {
   set pattern(pattern) { this._pattern = pattern; }
 
   static effectiveType(handleType, connections) {
+    let variableMap = new Map();
+    // It's OK to use _cloneWithResolutions here as for the purpose of this test, the handle set + handleType 
+    // contain the full set of type variable information that needs to be maintained across the clone.
+    let typeSet = connections.filter(connection => connection.type != null).map(connection => ({type: connection.type._cloneWithResolutions(variableMap), direction: connection.direction}));
+    return TypeChecker.processTypeList(handleType ? handleType._cloneWithResolutions(variableMap) : null, typeSet);
```

Comment:
This can probably be cleaned up now:
https://github.com/PolymerLabs/arcs/blob/master/runtime/recipe/type-checker.js#L27

Reply:
Actually I want to take another look at that copy, it's possibly incorrect...

Reply:
Cleaned up.